src/components/ha-date-range-picker.ts

```diff
@@ -254,21 +254,37 @@ export class HaDateRangePicker extends LitElement {
   }
 
   private _applyDateRange() {
+    let start = new Date(this._dateRangePicker.start);
+    let end = new Date(this._dateRangePicker.end);
+
+    if (this.timePicker) {
+      start.setSeconds(0);
+      start.setMilliseconds(0);
+      end.setSeconds(0);
+      end.setMilliseconds(0);
+
+      if (
+        end.getHours() === 0 &&
+        end.getMinutes() === 0 &&
+        start.getFullYear() === end.getFullYear() &&
+        start.getMonth() === end.getMonth() &&
+        start.getDate() === end.getDate()
+      ) {
+        end.setDate(end.getDate() + 1);
+      }
```

Comment:
This behavior is a little ambiguous. 

Previously we had if you pick the date of August 7th, start time 12:00 AM, end time 12:00 AM, we would advance the final date to August 8th, 12:00AM. I think this kind of makes sense, like you're likely trying to view the full day.

But we also had if you pick a date range like August 5th - August 6th, start time 12:00 AM, end time 12:00 AM, we would advance the end date to August 7th, 12:00 AM. That doesn't _really_ feel like the right thing to do? 

So I kept this behavior for only if the start and end date is the same, but I'm not sure if even that makes sense.